### ISSUE TITLE
🚀 Feature: overload Feature for Storage.CreateFile for using Blob instead of File

### ISSUE THREAD

**Issue by trashcoder** [appwrite/appwrite#5748]
### 🔖 Feature description

The method createFile of the class Storage can only work with the File parameter. While using react-native it would be easier to use blob instead of File, because react-native doesn't support the class "File".

### 🎤 Pitch

In my use case, within react-native there is no class "File", so i can't use the storage.createFile method to upload my files. Instead i have to use a blob and upload it via REST api. Also to slice the blobs into chunks makes it even harder to get a consistent file upload. With a blob feature in createFile the whole upload progress would be much easier.

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by joeyouss** [comment]
Hi @trashcoder , thank you for opening this. This seems like a good feature to have, hence we have opened it for community contributions.

**Comment by adarshjhaa100** [comment]
Hi, Can I work on this issue if nobody has picked it up?

**Comment by FreSauce** [comment]
hey @stnguyen90 , can you assign me this issue?

**Comment by stnguyen90** [comment]
@trashcoder, how are you getting your file that you want to send to Appwrite?

**Comment by trashcoder** [comment]
I get the file from react-native-vision-camera as uri. But because createFile can't handle uri's and react-native can't handle the File-Class a blob would be a good compromise.

**Comment by stnguyen90** [comment]
@trashcoder, thanks for the insight! As mentioned over [here](https://github.com/appwrite/sdk-for-web/issues/27#issuecomment-1710894043), React Native expects a File in FormData to look like:

```javascript
{
      uri: uri,
      name: filename,
      type: fileType,
}
```

One thing we need to figure out is how to make it work on Web and React Native with the same function. The function also needs to be able to handle chunked uploads for files bigger than 5MB. 🧐

**Comment by stnguyen90** [comment]
@adarshjhaa100, @FreSaucek, do you understand the [requirements](https://github.com/appwrite/appwrite/issues/5748#issuecomment-1720322803) and have an idea about how to implement this?

**Comment by FreSauce** [comment]
Currently, I don't have any idea for chunked uploads, will have to look into it more.

**Comment by trashcoder** [comment]
I think you can look at the solution of "createFile" with File as parameter and adapt it to blob and filetype.

**Comment by adarshjhaa100** [comment]
@stnguyen90 Yes I do have some idea on how chunk uploads are to be implemented. Using `fetch()` with the header `multipart/formdata` would take care of uploads in [react-native](https://reactnative.dev/docs/network?language=typescript#making-requests)  in a similar way like its done in [web-sdk](https://github.com/appwrite/appwrite/blob/master/public/sdk-web/services/storage.ts). I think same mechanism of looping over chunked uploads could be implemented in react native.

I could do some POC on this.

**Comment by rose2221** [comment]
hey, can i work on this issue?

**Comment by stnguyen90** [comment]
Closing as:

1. we have a [react native SDK](https://github.com/appwrite/sdk-for-react-native) now
2. the node sdk supports [File](https://github.com/appwrite/sdk-for-node/blob/74e8afb10cc80d83005b423dbf5b4b954f074c16/src/services/storage.ts#L277) and [Blob](https://github.com/appwrite/sdk-for-node/blob/74e8afb10cc80d83005b423dbf5b4b954f074c16/src/inputFile.ts#L6-L9)
